### ISSUE TITLE
Feedback about the new pull request comment

### ISSUE THREAD

**Issue by kylemann** [codecov/feedback#8]
### Thanks for dropping by! 👋

We've been iterating and updating the layout, summary, and copy of the pull request comment. 

* What do you think about the changes?
* What would you'd like to see improved?
* What are challenges you've experienced with the report?
* How do you use the comment today?
* How could the report better help your workflow?
* Any general thoughts you'd like to share!

We greatly appreciate your time and thoughts - looking forward to hearing from you ❤

Codecov team

_This issue is intended to share and collect feedback about the tool. If you have support needs or questions, please see our[ support page](https://codecov.freshdesk.com/support/home)._

**Comment by Codecovdesign** [comment]
@TechMaz thank you for your feedback.
We are actually experimenting with a new layout, see below. It prioritizes the high level summary and then collapses the other reporting data (impacted files, table, etc).

Would love to hear your feedback 🙏 


<img width="1440" alt="new_comment" src="https://user-images.githubusercontent.com/87861036/156568566-4b4d1357-05d8-4418-9fcb-406761c2daa6.png">

**Comment by openrefactory** [comment]
@eliatcodecov 
Please watch this issue we have open this issue one week ago still don't get any respond yet.
# https://github.com/httpie/httpie/pull/1434

**Comment by daniellockyer** [comment]
I love it! It's so much cleaner and less obtuse <3

**Comment by unmade** [comment]
I'm not sure how that's possible but pull request form shows me wrong coverage result - it says my pull request will decrease project coverage, although I'm confident my changes won't not affect coverage at all. I don't see that problem in the codecov web application. 

I also compared coverage.xml files uploaded to codecov for corresponding commits and they're identical. 

The pull request with wrong form:
https://github.com/unmade/shelf-back/pull/97

Full report at codecov:
https://app.codecov.io/gh/unmade/shelf-back/pull/97

<img width="1668" alt="Screenshot 2022-10-05 at 22 22 19" src="https://user-images.githubusercontent.com/9870176/194149612-74584483-a3ee-442c-a16e-c578673f19ae.png">

And here's the coverage files for corresponding commits:
- HEAD: [a1e3d35.txt](https://github.com/codecov/Codecov-user-feedback/files/9718578/a1e3d35.txt)
- BASE: [1baf419.txt](https://github.com/codecov/Codecov-user-feedback/files/9718576/1baf419.txt)

**Comment by nitram509** [comment]
Hi,

thanks for providing codecov tests to the Open Source community for free.
After adding some tests, I found your report reported lower coverage than before, when no test was there at all.
See https://github.com/argoproj/argo-rollouts/pull/2303#issuecomment-1272410769

I do not trust the codecov report, because why it's going down when I literally create the first test file in this package?

PS: I just measured the test coverage of the new file created, this is 87% (according to IntelliJ).

**Comment by KnorpelSenf** [comment]
This may sound like a nitpick but the coverage diff has a sign error in the GitHub comment that the bot is posting. It says "coverage decreases by `-3%`" but decreasing by a negative amount would mean increasing. Perhaps it should say "coverage decreases: `-3%`" to make it more accurate?

**Comment by ssb22** [comment]
On [this pull request](https://github.com/huashengdun/webssh/pull/301) I simply added some lines to the documentation (README.md and README.rst), but CodeCov warned that code coverage had been reduced by my change.  It would be nice if obvious documentation files didn't count.  Thanks.

**Comment by mamil** [comment]
Hi，

On this [PR](https://github.com/cloudwego/hertz/pull/403), I just copy an old case and then mod one function call, no tests were removed, and the test coverage decreased.

**Comment by sayo96** [comment]
I'm unable to view the coverage report like before on the PR's. Codecov is such a big mess rn.

**Comment by Codecovdesign** [comment]
> "coverage decreases by -3%"...Perhaps it should say "coverage decreases: -3%" to make it more accurate?

Thank you, @KnorpelSenf that makes sense - will update text clarification in next iteration.

**Comment by PaulRBerg** [comment]
Is it possible to disable the automatic comments?

We'd like to keep using the Codecov GitHub app (so that our `codecov.yml` file gets synced with Codecov) but not receive the PR comments.

**Comment by slifty** [comment]
Is it possible to manually request that codecov re-run the report?  I've found the most spotty aspect of codecov in our development workflow to be when we force push to a branch with new test coverage but codecov doesn't update the statistics.

It would be great to be able to basically say "@codecov rerun" or something along those lines in the PR.

**Comment by ntindle** [comment]
Hey! Over at https://github.com/Significant-Gravitas/Auto-GPT/ we are running into some serious problems with CodeCov. We tried reaching out to a rep and maybe getting some of it worked out and a demo, but they no-showed our meeting. We really liked CodeCov, but it's currently broken and denying all PRs. I don't particularly want to swap providers but we have hundreds of people asking us why CodeCov is broken and we can't find answers. Would love a reply as soon as you can

**Comment by borislavnnikolov** [comment]
What about the StateHasChanged implementation for the other components, like MudTabs and etc. ?

**Comment by webbnh** [comment]
The pull request comment seems good to me (I'm just starting with Codecov), but there seems to be a bootstrap problem which I describe [here](https://app.codecov.io/gh/feedback/p/treat-base-commit-as-empty-if-it-cannot-be-found):  when there is nothing to compare to (because the base commit doesn't have any code in it yet), it's treated as a error situation instead of as though all the changes are new code (which...they are 😁).

Once my PR is merged, we'll presumably never encounter the situation again, so perhaps it's not real important, but fixing it would improve the user's on-boarding experience, and it doesn't seem like it would be that hard to address.

**Comment by iselo** [comment]
Let's do the math!)
![Screen Shot 2023-07-15 at 3 59 46 PM](https://github.com/codecov/feedback/assets/50623030/3b46c807-f971-47dd-ac69-ea0045723873)

**Comment by KnorpelSenf** [comment]
Those are rounded numbers, so it could have gone down by 0.008 from 99.614 to 99.606 which would be rounded to 0.01 from 99.61 to 99.61.

**Comment by webbnh** [comment]
Explicit rounding aside, floating point math is much harder than it seems.  (Any time you see a decimal point in the middle of the number, assume that anything to the right of it is "an estimate"!  😁)

**Comment by Gottox** [comment]
Thanks for your great service!

I'm not sure if I'm doing smth wrong, but I get the following warning on my personal (!) project:

> ❗ Your organization is not using the GitHub App Integration. As a result you may experience degraded service beginning May 15th. Please [install the Github App Integration](https://github.com/apps/codecov) for your organization. [Read more](https://about.codecov.io/blog/codecov-is-updating-its-github-integration/?utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Enno+T.+Boland).

I also enabled the app for this account:
![image](https://github.com/codecov/feedback/assets/1056976/bfb329f9-853e-4941-ac57-264b96985e89)

Is that a false positive?

**Comment by kylemann** [comment]
@Gottox thank you for raising this 🙏 . If the GitHub app is installed, this appears to be a bug and should not have been shown. I've created an bug issue and raised with the team.

**Comment by codecovdesign** [comment]
@Gottox thank you for raising this 🙏 . If the GitHub app is installed, this appears to be a bug and should not have been shown. I've created an bug issue and raised with the team.

**Comment by yuekui** [comment]
We've already setup comment without `feedback`:
![image](https://github.com/codecov/feedback/assets/2561450/7d2663a9-bdc1-44fd-b984-0cffa26850a9)

But why we're still getting this `feedback` section?
![image](https://github.com/codecov/feedback/assets/2561450/f42d51bc-d886-41ab-a9ae-18053f02f62a)

**Comment by codecovdesign** [comment]
@yuekui thank you for surfacing this 🙏. We investigated it and appears to have been a bug that is now fixed.
Let me know if you're still seeing the issue!

**Comment by karfau** [comment]
![image](https://github.com/codecov/feedback/assets/135657/4c174ea5-ba6a-411e-835e-6d2a8e88049d)
I took me a while to understand that this means the current run that uploads the report has not completed yet.

I would suggest to make this more obvious by changing the wording to something like

> This the last uploaded report is for commit 1234567 but the latest commit for this PR is 4567890, this comment will be updated as soon as the report for that commit has been uploaded.

**Comment by zimeg** [comment]
👋 Hey there, big fan of PR comments and y'alls coverage checks in general! Really appreciate this place for feedback!


<details>
<summary>A recent change might have moved the `diff` into a details section with `files` being shown at the top level
</summary>

<img width="734" alt="patch" src="https://github.com/codecov/feedback/assets/18134219/c8c25dd0-3f87-4355-ba83-ba68e7a85f81">

</details>

Previously only the `diff` was shown with the same header and footer, without the specifics on `files`. I preferred that approach since it focused more on improving coverage in the patch instead of missing coverage[^1].

I found [a section in the docs](https://docs.codecov.com/docs/pull-request-comments#layout) on changing this layout and was wondering if the `comment` example[^2] is accurate with the dropdown above? The `codecov.yml` used in the comment's project has no `layout` section at the moment.

Also wondering how adding `condensed_` changes the comment layout? I'll probably try out a few different options soon but it's not super clear to me right now. Thanks for all of the great features! :raised_hands:

[^1]: 😉 💚 https://about.codecov.io/blog/the-case-against-100-code-coverage/
[^2]: Small note, there's an extra space before the `hide_project_coverage` field

**Comment by rohan-at-sentry** [comment]
> 👋 Hey there, big fan of PR comments and y'alls coverage checks in general! Really appreciate this place for feedback!
> 
> A recent change might have moved the `diff` into a details section with `files` being shown at the top level
> Previously only the `diff` was shown with the same header and footer, without the specifics on `files`. I preferred that approach since it focused more on improving coverage in the patch instead of missing coverage[1](#user-content-fn-1-1deb3b31f8bca5f1115a73c9f726cb8c).
> 
> I found [a section in the docs](https://docs.codecov.com/docs/pull-request-comments#layout) on changing this layout and was wondering if the `comment` example[2](#user-content-fn-2-1deb3b31f8bca5f1115a73c9f726cb8c) is accurate with the dropdown above? The `codecov.yml` used in the comment's project has no `layout` section at the moment.
> 
> Also wondering how adding `condensed_` changes the comment layout? I'll probably try out a few different options soon but it's not super clear to me right now. Thanks for all of the great features! 🙌
> 
> ## Footnotes
> 1. 😉 💚 https://about.codecov.io/blog/the-case-against-100-code-coverage/ [↩](#user-content-fnref-1-1deb3b31f8bca5f1115a73c9f726cb8c)
> 2. Small note, there's an extra space before the `hide_project_coverage` field [↩](#user-content-fnref-2-1deb3b31f8bca5f1115a73c9f726cb8c)

@zimeg thanks for the detailed feedback here 👍🏽 

You can bring back the older view by adding the following to your `codecov.yml`

```
layout: " header, diff, files, footer"
``` 

However, since you're open to you, I'd encourage experimenting with the options a little bit, there's a lot of good stuff in the new changes that I'm sure you'd benefit from.

Adding `condensed_` accomplishes the following for each element


- `condensed_header` : updated verbiage to be more concise.
- `condensed_diff` : is more verbose - giving you coverage "diff" between current and last commit for which we have coverage
- `condensed_files` : is now furled by default, which is largely driven by feedback we've historically received about the PR comment being large

**Comment by shlomi-noach** [comment]
In this report, `codecov` tells me that there is 1 line missing coverage _in my change_:

![image](https://github.com/codecov/feedback/assets/2607934/65e90aa0-936f-4610-af17-b93b831e623f)

However, clicking `1 missing`, or any other link, only leads me to the full coverae breakdown of `throttler.go`, nothing to show me what specific 1 line was missing coverage _in my change_. 

Link to OSS `vitessio/vitess` PR comment: https://github.com/vitessio/vitess/pull/14971#issuecomment-1895210210

**Comment by FawziEid** [comment]
vitessio/vitess

**Comment by donhcd** [comment]
I'm getting a failing symbol ❌ in my PR checks 
<img width="861" alt="image" src="https://github.com/codecov/feedback/assets/902007/db920ee2-9217-4959-b33a-695cc6d42068">
even though the comment on the same PR has a ✅ on it
<img width="902" alt="image" src="https://github.com/codecov/feedback/assets/902007/b54f8b03-423e-4e5d-9833-cb0e5115345d">

ever since we introduced codecov to our project, probably 75+% of our PRs have had ❌s on them, probably due to this class of issue. In this specific case, my PR only affects github actions files -- why is codecov ❌ing me?

**Comment by thomasrockhu-codecov** [comment]
@donhcd can you provide a link to your PR or a commit SHA?

**Comment by nhorton** [comment]
Meta request: The way this solicitation of feedback is linked in the comment means that anything that syncs comments to places with previews (like Slack) results in a giant blob about this feedback request in those channels. If you made this go through a link shortener or such to break the preview, it would be really nice.

**Comment by naik-aakash** [comment]
The codecov report is bit inconsistent with on different runs for same test set.

**Comment by HarelM** [comment]
I'm not sure if this is only me reading this incorrectly, but I'm having a hard time truly understanding the difference between the PR comment and the report itself, they don't seem to be aligned:
Relevant PR: https://github.com/maplibre/maplibre-gl-js/pull/3645
Screenshot of current comment:
![image](https://github.com/codecov/feedback/assets/3269297/2fd2a7f2-85ef-4f1b-8b47-ced3c4ad0f98)
Screenshot of the report in codecov:
![image](https://github.com/codecov/feedback/assets/3269297/f39ac7f9-1c89-4735-a33c-553917528487)

The number of missed lines is not the same.
The patch percentage is not the same.
I'm not sure which one of them should I trust... :-/

**Comment by codecovdesign** [comment]
@HarelM thank you for the detailed report 🙏 
I'm not sure what's happening here, but the results are clearly different. 
Our team is investigating it, created a bug report: https://github.com/codecov/engineering-team/issues/1321
Thanks again for sharing!

**Comment by dabrahams** [comment]
Is there a way to make codecov not send the report when all is well?  I really only want to hear about failures.  Thanks!

**Comment by HarelM** [comment]
It's also worth noting the following experience:
The above library that I linked is sending multiple coverage files since some of the tests run on different workers in parallel.
Codecov will create a coverage comment when the first file al finishes processing, so I get a mail notifying me with wrong coverage. In most cases this coverage comment will be updated although I've seen cases where it didn't.
I would consider posting the coverage comment only when all the runners complete.

I know our use case might be complicated, but coverage is very important to our project, I've done a lot of tweaking to be able to measure coverage correctly.

**Comment by codecovdesign** [comment]
> Is there a way to make codecov not send the report when all is well? I really only want to hear about failures.

@dabrahams there is [this configuration](https://docs.codecov.com/docs/common-recipe-list#only-comment-on-coverage-changes) that will only show if ANY changes occur. That doesn't quite acheive your ask though. cc'ing @thomasrockhu-codecov @drazisil-codecov are you familiar with another configuration to resolve this?

Created an issue to look at this more / adding ability to show only with negative reporting: https://github.com/codecov/engineering-team/issues/1350; please add any thoughts/comments you have.

**Comment by codecovdesign** [comment]
> I get a mail notifying me with wrong coverage

@HarelM it may help to apply[ after_n_buid here](https://docs.codecov.com/docs/pull-request-comments#after_n_builds), do you have this set? It will wait until that many reports are complete to update the comment.

**Comment by HarelM** [comment]
Ah, didn't know about this option, I'll try it out, thanks!

**Comment by HarelM** [comment]
@codecovdesign I don't see the specified flag in the github action parameters here:
https://github.com/codecov/codecov-action
Am I missing anything?

**Comment by codecovdesign** [comment]
@HarelM if updating flags you'll want to do this in the codecov.yaml https://docs.codecov.com/docs/flags#step-2-flag-management-in-yaml

**Comment by Eianaftyio** [comment]
.develop

**Comment by HarelM** [comment]
@codecovdesign is there a way to do it without introducing a whole new file to my repo in order to send a single flag to codecov? Is it possible to support the most common flag (or better yet all of them) in the github action configuration? I'm managing most of my CI configuration in this file anyway, it would be great to be able to simply add `with: something` or add some environment variables that the action will take into consideration.

**Comment by Eianaftyio** [comment]
with:Certainly, here are example reply comments that could be associated with the discussions around improving the Ethereum Request for Comments (ERCs) project and community:

1. "Thanks for bringing up this important topic. Building a thriving community around ERCs is crucial for their widespread adoption and effectiveness. Let's brainstorm ways to emulate the success of the EIPs community."
2. "I agree that fostering a strong sense of community is essential for the ERCs project. One idea could be to organize regular community events such as hackathons or workshops to encourage collaboration and engagement among contributors."
3. "In addition to events, we should also focus on improving the documentation and guidelines for ERC submissions. Clearer guidelines will make it easier for authors to create high-quality proposals and for reviewers to provide constructive feedback."
4. "It might also be beneficial to establish a more structured review process for ERCs, similar to the EIPs process. This could involve defining specific stages for proposal review and incorporating feedback from a diverse group of community members."
5. "Building bridges with other Ethereum-related projects and communities could also help enhance the visibility and impact of ERCs. Collaborations with developer groups, research institutions, and industry partners can bring fresh perspectives and expertise to the table."
6. "Let's ensure that the ERCs repository is welcoming to contributors of all backgrounds and skill levels. Providing mentorship opportunities and creating a supportive environment for newcomers can help grow the community in a sustainable way."
7. "Lastly, regular communication and transparency are key to maintaining a healthy community. We should strive to keep channels open for discussion, address concerns promptly, and celebrate achievements to foster a sense of belonging and pride among ERCs contributors."

**Comment by Eianaftyio** [comment]
With:Certainly, here are example reply comments that could be associated with the discussions around improving the Ethereum Request for Comments (ERCs) project and community:

1. "Thanks for bringing up this important topic. Building a thriving community around ERCs is crucial for their widespread adoption and effectiveness. Let's brainstorm ways to emulate the success of the EIPs community."
2. "I agree that fostering a strong sense of community is essential for the ERCs project. One idea could be to organize regular community events such as hackathons or workshops to encourage collaboration and engagement among contributors."
3. "In addition to events, we should also focus on improving the documentation and guidelines for ERC submissions. Clearer guidelines will make it easier for authors to create high-quality proposals and for reviewers to provide constructive feedback."
4. "It might also be beneficial to establish a more structured review process for ERCs, similar to the EIPs process. This could involve defining specific stages for proposal review and incorporating feedback from a diverse group of community members."
5. "Building bridges with other Ethereum-related projects and communities could also help enhance the visibility and impact of ERCs. Collaborations with developer groups, research institutions, and industry partners can bring fresh perspectives and expertise to the table."
6. "Let's ensure that the ERCs repository is welcoming to contributors of all backgrounds and skill levels. Providing mentorship opportunities and creating a supportive environment for newcomers can help grow the community in a sustainable way."
7. "Lastly, regular communication and transparency are key to maintaining a healthy community. We should strive to keep channels open for discussion, address concerns promptly, and celebrate achievements to foster a sense of belonging and pride among ERCs contributors."

**Comment by Eianaftyio** [comment]
Lots of love for you guys ❤️

**Comment by Eianaftyio** [comment]
.diff

**Comment by HarelM** [comment]
Another issue with the report: the following PR never got a comment with coverage:
https://github.com/maplibre/maplibre-gl-js/pull/3963
It started as a draft PR, then I found out there's problem with the token in 3, so I added a token and upgraded to 4 just to see that it still doesn't work, also moved it from draft to regular, but it still doesn't show the comment. I do see it in codecov.io site...
Also, there's another bug that's preventing from getting accurate report due to the fact that it takes too many files (not the files I specifically requested):
The following is the relevant bug:
https://github.com/codecov/codecov-action/issues/1354
One thing is not getting a PR comment, the other is incorrect data, which is far worse and I think can be considered a critical bug.

**Comment by codecovdesign** [comment]
@HarelM sorry missed [this one](https://github.com/codecov/feedback/issues/8#issuecomment-2017546167); @drazisil-codecov and I were just looking at it and were unsure exactly. would you be up for a call? if so, shoot me an email at kyle.mann@sentry.io and will find a time. curious to learn more about your [use case](https://github.com/codecov/feedback/issues/8#issuecomment-1982591213) too.

**Comment by HarelM** [comment]
I've sent a mail to see when can we meet.
Another input I just got from a fellow maintainer about the need for a token when using version 4:
When someone opens a PR from a fork (the most common option for our open-source project) the codecov secret is not available (see here: https://github.com/orgs/community/discussions/50161).
This will prevent me from seeing how the changes in the code affect the code coverage.
The only option I can think of (since I'm not sure what the codecov token actually guards) is placing the token inside the github actions yml file so it will be publicly available. Not a good solution, but I can't think of a simpler one (there is a hack in the linked discussion, but I think it's even worse in terms of DX, probably better in terms of security).
In any case, I'd love to sit and chat about our use case, I don't think it's an edge case, but we are talking coverage seriously.

**Comment by jiekun** [comment]
Hi I found codecov comment with incorrect base commit for my Pull Request.

See [here](https://github.com/VictoriaMetrics/VictoriaMetrics/pull/6124#issuecomment-2060275644)

I'm checking out new feature branch from the latest master commit. But codecov's comments in all my Pull Request said the base commit is 8aaa828ba3f49a43cb8de89a0bcae160f7988c14 and posting a wrong project coverage.
(The patch coverage is high and somehow the project coverage is decreasing)

May I know the root cause of this issue and somehow fix it on my side?

**Comment by gmlewis** [comment]
In the [go-github](https://github.com/google/go-github/pull/3156#issuecomment-2083851737) repo, we consistently see messages like:

> Report is 46 commits behind head on master.

without any explanation as to why this is (or how to fix it). As a result, the codecov report is not useful.

**Comment by superlopuh** [comment]
Some time in the last few weeks, the reporting of coverage by lines with diffs stopped working, as in this PR: https://github.com/xdslproject/xdsl/pull/2540
![Screenshot 2024-05-05 at 17 04 44](https://github.com/codecov/feedback/assets/910320/55a069b0-faef-45c9-8b8f-4c1807374608)
Clearly not all lines are covered by tests, and the additional info shows this, but the message disagrees. I am also not able to see the diff when clicking through to codecov.

**Comment by mortonjt** [comment]
Hi, thank you for providing this tool.  We have been using this for [scikit-bio](https://github.com/scikit-bio/scikit-bio) 

However, we have been recently having issue with codecode. Some files are mysteriously reporting little coverage, and it is not clear to us, since we have tried to create unittests that cover all of the tests (see screenshot and pull request in question [here](https://github.com/scikit-bio/scikit-bio/pull/2021))
 
![Screenshot from 2024-05-13 11-42-39](https://github.com/codecov/feedback/assets/4184797/96c93f42-cf0f-4d92-b764-254a72168796)

The only culprit that I can think of is a misconfiguration in the [.codecov.yml](https://github.com/scikit-bio/scikit-bio/blob/main/.codecov.yml) or [.coveragerc](https://github.com/scikit-bio/scikit-bio/blob/main/.coveragerc) files.

Feedback would be greatly appreciated. If there is a more appropriate forum to follow up, we'll be happy to move this discussion there

**Comment by gward35** [comment]
Is there a way to customize the comment output so that we see top level coverage file coverage for branches/functions etc and not just total coverage? e.g. 

```
Codecov Report

// all the typical stuff
...
Statements: 71.86%
Branch: 65.31%
Functions: 71.19%
Lines: 71.86%
```

I dont mind the patch showing the diff but at the end I also just want a plain output of the coverage in these respective buckets

**Comment by emmadesilva** [comment]
The only reason I have Codecov is to see if a PR will add or remove coverage, as such I only care about that metric. Please default to have this open window open, or at least display the red/green diff in the main body! It used to be a lot better.

![image](https://github.com/codecov/feedback/assets/95144705/a1333168-f1cb-422e-b299-a873e3c8289e)

**Comment by codecovdesign** [comment]
thank you @caendesilva! i opened this issue to capture the feedback: https://github.com/codecov/engineering-team/issues/2022. any thoughts on the update? it adds the project change to the top level summary for quick at-a-glance review.

**Comment by emmadesilva** [comment]
> thank you @caendesilva! i opened this issue to capture the feedback: [codecov/engineering-team#2022](https://github.com/codecov/engineering-team/issues/2022). any thoughts on the update? it adds the project change to the top level summary for quick at-a-glance review.

Thanks! Left a comment there.

**Comment by drazisil** [comment]
@codecovdesign I feel like zero uploaded HEAD reports should be showing as a missing head report, wdyt? :D

https://github.com/rustymotors/rusty/pull/32#issuecomment-2252489892 

![Image](https://github.com/user-attachments/assets/00ae28b9-44d8-4914-bdb7-bf4c4894d12d)

**Comment by karfau** [comment]
Here is another case where the report seems to calculate something strange: https://github.com/xmldom/xmldom/pull/698#issuecomment-2254651688



![Image](https://github.com/user-attachments/assets/ec5d5663-2196-4539-b077-78194584fb7d)

Not sure how it is calculated, but when 6 lines and one branch are being removed, they can of course also no longer be hit?

Or am I misunderstanding something?

I even added/extended tests in that commit.

**Comment by drazisil-codecov** [comment]
@karfau Not trying to discount your feedback. Is https://docs.codecov.com/docs/removed-code-behavior of help in your case?

**Comment by karfau** [comment]
@drazisil-codecov if I understand this documentation correctly, the default is `adjust_base` which shouldn't fail the check, right?
And it looks as if I have not configured it: https://github.com/xmldom/xmldom/blob/master/codecov.yml

**Comment by drazisil-codecov** [comment]
@karfau I agree with you, after taking a look. @rohan-at-sentry is this a bug, or some part of the system that needs better docs?

https://app.codecov.io/gh/xmldom/xmldom/pull/698?dropdown=coverage&src=pr&el=continue&utm_medium=referral&utm_source=github&utm_content=checks&utm_campaign=pr+comments&utm_term=xmldom is the only changed file, and it shouldn't fail project, given it's 100% covered.

**Comment by rohan-at-sentry** [comment]
@karfau - we'll look into why `adjust_base` isn't quite working well in your scenario. In the meantime, you can set the `removed code behavior` to allow a fully covered patch that removes lines to be accepted. You can edit your codecov.yml with the following setting


```
coverage:
  status:
    project:
      default:
        target: auto
        removed_code_behavior: fully_covered_patch 
```

**Comment by drazisil-codecov** [comment]
Created https://github.com/codecov/feedback/issues/463 to investigate

**Comment by moi90** [comment]
I get comments by codecov bot on PRs that were already merged a long time ago, like this one: https://github.com/morphocut/morphocut/pull/105#issuecomment-2316571448

**Comment by rohan-at-sentry** [comment]
@moi90 thanks for raising. It seems like it's related to some recent improvements we made to the notification service. We're tracking this here (https://github.com/codecov/feedback/issues/497) if you want to follow along.

**Comment by tiltowait** [comment]
codecov is great, but the report is always telling me to install the app, even though I'm pretty sure it is? Do I need to install it on *all* repos to suppress this message? I only need it on the one for now.

**Report screenshot:**
![Image](https://github.com/user-attachments/assets/2d8fb419-f63f-4972-ac0a-ddf630efed28)

**Github settings screenshot:**
![Image](https://github.com/user-attachments/assets/40e9f6c6-cdd0-41ab-8b85-c129284bf0b4)

**Comment by 19RachelZoieJovie97** [comment]
Good

**Comment by kcrwfrd** [comment]
Codecov seems to prematurely fail on our PRs, and then shortly afterwards it updates with a pass instead.

**Comment by eliatcodecov** [comment]
@kcrwfrd do you upload multiple coverage reports per commit? If so, codecov is likely failing because it's reporting on the first processed coverage upload, then updating the PR comment and statuses as new uploads are processed. 

There are a couples ways to prevent this behavior, but the easiest is to use `after_n_builds` ([docs](https://docs.codecov.com/docs/notifications#preventing-notifications-until-after-n-builds)) in your codecov.yml. If you upload the same number of coverage reports per commit, just set `after_n_builds` to however many reports you upload and codecov will not report at all until that number of uploads are processed.  

If you upload a differing number of reports on every commit, perhaps due to a cached or iterative test setup, you can still prevent this issue, it's just more involved. You can use the manual trigger ([docs](https://docs.codecov.com/docs/codecovyml-reference#codecovnotifymanual_trigger)) functionality of the codecov-cli to tell codecov when you've finished uploading. Honestly, though, if you upload the same number of reports on every commit, just use `after_n_builds`, it's much easier to adopt and maintain.

**Comment by dexxxxca** [comment]
is not easy 
do you upload multiple coverage reports per commit? If so, codecov is likely failing because it's reporting on the first processed coverage upload, then updating the PR comment and statuses as new uploads are processed.

**Comment by pboling** [comment]
Codecov has stopped working on all my repos.  The report upload works in GHA without any errors, but claims errors in your UI, and those errors are inscrutable.  All my repos show 0%, so I have begun the arduous task of ripping codecov out of my 100+ projects.

Example: https://github.com/oauth-xx/oauth2/pull/652#issuecomment-2912103674

UPDATE - Working with the codecov team, we've diagnosed several bugs in CodeCov. Some have been fixed, and others are still open. It is working for some of my repos now!

**Comment by elringus** [comment]
I can't even login to the Codecov dashboard. It just returns plain `403 Forbidden`. RIP

**Comment by agourdin** [comment]
Your reporting interface is really hard to share with AI agents. I can't take a good screenshot of the whole page because you only render what's visible and loaded it seems. And there's no plain text output that allows for copy and pasting. It would be VERY VERY helpful if we had some representation of the report that was easily shareable with AI agents, preferably as a link from the comment.

Thank you!

**Comment by rohan-at-sentry** [comment]
@agourdin - out of curiosity, can I ask what you're looking to solve using AI here? Are you looking to generate missing tests or other usecases? 

Context - we're working on some features in this area and I was wondering if there is already a solution here for you

**Comment by Phamphuluan79** [comment]
3AZHcgLnJL5C5xKo33mspyHpQX7x4H5bBw

**Comment by Saad4343** [comment]
[[Upload-from-mobile-1761188002.Book.3.1.xlsx](https://github.com/user-attachments/files/23153231/Upload-from-mobile-1761188002.Book.3.1.xlsx)]()

**Comment by Saad4343** [comment]
**_[]()_**

**Comment by gf220gt-ctrl** [comment]
GG ty co ci chodzi to chodzi.

**Comment by CscottT** [comment]
All CLEAR!

**Comment by garciaerubiel42-cmyk** [comment]
Am i still in sodo mode and yeah thanks again for the oppornity to expirince my tool once more and over its been really fun being with all you and seeing how thos will end up as ŵhole. As far a improvent we might consider bringing it dow a knotch to let our comperiter keep up l. As far as the coverage ive been told it increases it performance and coverage.

**Comment by RainbowMango** [comment]
Don't know why we got a comment on a PR that was merged 3 years ago. Is there something wrong?
https://github.com/karmada-io/karmada/pull/2680#issuecomment-4165209812

**Comment by 304NotModified** [comment]
> Don't know why we got a comment on a PR that was merged 3 years ago. Is there something wrong? [karmada-io/karmada#2680 (comment)](https://github.com/karmada-io/karmada/pull/2680#issuecomment-4165209812)

We have the same issue. It's not a big deal, but it confusing and there are multiple of these (unneeded) comments already. 

See https://github.com/NLog/NLog.Extensions.Logging/pull/412#issuecomment-4175214846

**Comment by imnasnainaec** [comment]
Same here:

- https://github.com/sillsdev/TheCombine/pull/748#issuecomment-4174269273
- https://github.com/sillsdev/TheCombine/pull/2719#issuecomment-4177513904

**Comment by jsoref** [comment]
WTF.

- https://github.com/jsoref/act/pull/69#issuecomment-4346250415
- https://github.com/jsoref/act/pull/47#issuecomment-4353177671
- https://github.com/jsoref/act/pull/45#issuecomment-4349609446
- https://github.com/jsoref/act/pull/60#issuecomment-4365646800
- https://github.com/jsoref/act/pull/56#issuecomment-4366500058

**Comment by 304NotModified** [comment]
FYI I've emailed Codecov (hello-codecov@sentry.io) for this issue.

**Comment by jsoref** [comment]
@thomasrockhu-codecov is anyone from @codecov attached to this discussion?

**Comment by drazisil-codecov** [comment]
Taking a look...

**Comment by drazisil-codecov** [comment]
Hi folks! Sorry for the delay on this. Quick question for my notes: do any of you have the Codecov GitHub app installed on the affected repos?

  Our working theory: when those PRs were closed or merged, Codecov never received the pull_request webhook (which is how we learn a PR's state has changed). Without that, our database still believed the PR was
  open.

  Codecov has a sync mechanism where loading a PR's coverage page triggers an async task to fetch the latest state and send updated status checks. This was designed as a safety net to catch cases where webhooks
  were delayed or dropped. Because of how that task is sequenced, the PR comment fires before we confirm the current state — meaning for these stale PRs, a comment went out before we realized they'd been closed.

  For the affected repos, it looks like someone loaded the Codecov coverage page for those PRs, which triggered the update.

  We've shipped a fix: https://github.com/codecov/umbrella/pull/910

**Comment by jsoref** [comment]
@drazisil-codecov 

https://github.com/jsoref/act/settings/installations

<img width="799" height="159" alt="Image" src="https://github.com/user-attachments/assets/79c559bc-747c-4b23-9b3a-971309d2d867" />

My guess is that I _never_ installed such an app, but I'm not going to make such a strong claim.

**Comment by abrahem79** [comment]
`/curl --request POST \
  --url 'https://api.tenderly.co/api/public/v1/account/{accountSlug}/project/{projectSlug}/environments' \
  --header 'X-Access-Key: <your-access-key>' \
  --header 'Content-Type: application/json' \
  --data '{
  "display_name": "Staging Mainnet",
  "slug": "staging-mainnet",
  "network_configs": [
    {
      "network_id": "1",
      "block_number": "latest",
      "chain_config_overrides": {
        "chain_id": "1"
      },
      "explorer_config": {
        "enabled": false,
        "verification_visibility": "bytecode"
      }
    }
  ]
}'`

**Comment by abrahem79** [comment]
> [``](n1. tomato     2. source     3. jump       4. windy      5. inspire
6. window     7. action     8. believe    9. lighting   10. deserve
11. lettuce   12. bicycle   13. mitten    14. exercise  15. school
16. guitar    17. kitchen   18. birds     19. decide    20. orange
21. special   22. glitter   23. blended   24. washer    25. comfort)
